.github/scripts/main/main.sh

```diff
@@ -121,35 +155,52 @@ if [ "$OPAM_TEST" = "1" ]; then
 
   # Compile and run opam-rt
   (set +x ; echo -en "::group::opam-rt\r") 2>/dev/null
-  opamrt_url="https://github.com/ocaml-opam/opam-rt"
-  if [ ! -d $CACHE/opam-rt ]; then
-    git clone $opamrt_url  $CACHE/opam-rt
-  fi
-  cd $CACHE/opam-rt
-  git fetch origin
-  if [ "$GITHUB_EVENT_NAME" = "pull_request" ] && git ls-remote --exit-code origin "$GITHUB_PR_USER/$BRANCH" ; then
-    BRANCH=$GITHUB_PR_USER/$BRANCH
-  fi
-  if git ls-remote --exit-code origin "$BRANCH"; then
-    OPAM_RT_BRANCH=$BRANCH
-  elif [ "$GITHUB_EVENT_NAME" = pull_request ] && git ls-remote --exit-code origin "$GITHUB_BASE_REF"; then
-    OPAM_RT_BRANCH=$GITHUB_BASE_REF
-  else
-    OPAM_RT_BRANCH=master
-  fi
-  if git branch | grep -q "$OPAM_RT_BRANCH"; then
-    git checkout "$OPAM_RT_BRANCH"
-    git reset --hard "origin/$OPAM_RT_BRANCH"
-  else
-    git checkout -b "$OPAM_RT_BRANCH" "origin/$OPAM_RT_BRANCH"
-  fi
+  prepare_project "ocaml-opam" "opam-rt"
 
-  test -d _opam || opam switch create . --no-install --formula '"ocaml-system"'
-  eval $(opam env)
-  opam pin $GITHUB_WORKSPACE -yn --with-version to-test
   # opam lib pins defined in opam-rt are ignored as there is a local pin
   opam pin . -yn --ignore-pin-depends
   opam install opam-rt --deps-only opam-devel.to-test
-  make || { opam reinstall opam-client -y; make; }
+  opam exec -- make || { opam reinstall opam-client -y; opam exec -- make; }
   (set +x ; echo -en "::endgroup::opam-rt\r") 2>/dev/null
 fi
+
+test_project () {
+  org=$1
+  project=$2
+  ignore_depends=$3
+  make_cmd=$4
+
+  ignore=""
+  if [ $ignore_depends -eq 1 ]; then
+    ignore="--ignore-pin-depends"
+  fi
+
+  (set +x; echo -en "::group::depends-$project\r") 2>/dev/null
+  prepare_project "$org" "$project"
+  set +e
+  opam pin . -yn $ignore
+  opam install "$project" --deps-only opam-client.to-test
+  opam exec -- $make_cmd || { opam reinstall opam-client -y; opam exec -- $make_cmd; }
+  code=$?
+  if [ $code -ne 0 ]; then
+    DEPENDS_ERRORS="$DEPENDS_ERRORS $project"
+  fi
+  set -e
+  (set +x ; echo -en "::endgroup::depends-$project\r") 2>/dev/null
+}
+
+if [ "$OPAM_DEPENDS" = "1" ]; then
+
+  DEPENDS_ERRORS=""
+  (set +x; echo -en "::group::depends\r") 2>/dev/null
+
+  test_project "ocaml-opam" "opam-publish" 0 "make"
+  test_project "AltGr" "opam-bundle" 0 "make"
+  test_project "ocamlpro" "opam-custom-install" 1 "dune build"
+
```

Comment:
I would also suggest adding `dune-release`, `odep`, `odig`, `odoc-driver`, `opam-check-npm-deps`, `opam-build`, `opam-test`, `opam-dune-lint`, `opam-graph`, `opam-package-upgrade`, `orb`, `opam-0install` and `topkg-care`

Reply:
It was package to test (some are failing in purpose as they have a dependency on opam < 2.2). We'll need to select what package to add here : they are blocking for the PR merge.
Ideally we would have another more light mechanism, one that sends the information as a comment in the PR for ex., like that we are aware without blocking the PR (for packages that we don't maintain).